sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -2404,6 +2409,19 @@ pi_result cuda_piKernelCreate(pi_program program, const char *kernel_name,
   }
 
   *kernel = retKernel.release();
```

Comment:
This line should be right before the return. This means if anything fails in the following code, we won't leak memory as the unique_ptr will stay as the owner.

Reply:
done